bot.ts

```diff
@@ -266,7 +305,7 @@ function PrintError(type, ...args: any[]) {
 	}
 }
 
-function GetActionErrorText(location, index, dataName) {
+function GetActionErrorText(location: string, index: number, dataName: string) {
```

Comment:
This looks right but very well may not be. please inform me if its incorrect. location is the sketchy one the others seems self explanatory.